src/tests/functional/shared_test_classes/src/single_op/non_max_suppression.cpp

```diff
@@ -0,0 +1,161 @@
+
+    ov::ParameterVector params {std::make_shared<ov::op::v0::Parameter>(params_type, ov::Shape(boxes_shape)),
+                                std::make_shared<ov::op::v0::Parameter>(params_type, ov::Shape(scores_shape))};
+
+    auto max_out_boxes_per_class_node = std::make_shared<ov::op::v0::Constant>(max_box_type, ov::Shape{}, std::vector<int>{max_out_boxes_per_class});
+    auto iou_thr_node = std::make_shared<ov::op::v0::Constant>(thr_type, ov::Shape{}, std::vector<float>{iou_thr});
+    auto score_thr_node = std::make_shared<ov::op::v0::Constant>(thr_type, ov::Shape{}, std::vector<float>{score_thr});
+    auto soft_nms_sigma_node = std::make_shared<ov::op::v0::Constant>(thr_type, ov::Shape{}, std::vector<float>{soft_nms_sigma});
+
+    auto nms = std::make_shared<ov::op::v5::NonMaxSuppression>(params[0],
+                                                               params[1],
+                                                               max_out_boxes_per_class_node,
+                                                               iou_thr_node,
+                                                               score_thr_node,
+                                                               soft_nms_sigma_node,
+                                                               box_encoding,
+                                                               sort_res_descend,
+                                                               out_type);
+
+    auto nms_0_identity =
+        std::make_shared<ov::op::v1::Multiply>(nms->output(0), ov::op::v0::Constant::create(out_type, ov::Shape{1}, {1}));
+    auto nms_1_identity =
+        std::make_shared<ov::op::v1::Multiply>(nms->output(1), ov::op::v0::Constant::create(params_type, ov::Shape{1}, {1}));
+    auto nms_2_identity =
+        std::make_shared<ov::op::v1::Multiply>(nms->output(2), ov::op::v0::Constant::create(out_type, ov::Shape{1}, {1}));
+    nms_0_identity->set_friendly_name("Multiply_0");
+    nms_1_identity->set_friendly_name("Multiply_1");
+    nms_2_identity->set_friendly_name("Multiply_2");
+    function = std::make_shared<ov::Model>(OutputVector{nms_0_identity, nms_1_identity, nms_2_identity}, params, "NMS");
+}
+
+void Nms9LayerTest::SetUp() {
+    InputTypes input_types;
+    InputShapeParams input_shape_params;
+    int max_out_boxes_per_class;
+    float iou_thr, score_thr, soft_nms_sigma;
+    op::v5::NonMaxSuppression::BoxEncodingType box_encoding;
+    op::v9::NonMaxSuppression::BoxEncodingType box_encoding_v9;
+    bool sort_res_descend;
+    ov::element::Type out_type;
+    std::tie(input_shape_params,
+             input_types,
+             max_out_boxes_per_class,
+             iou_thr,
+             score_thr,
+             soft_nms_sigma,
+             box_encoding,
+             sort_res_descend,
+             out_type,
+             targetDevice) = this->GetParam();
+
+    box_encoding_v9 = box_encoding == op::v5::NonMaxSuppression::BoxEncodingType::CENTER ?
```

Comment:
let's fix GPU instantiation instead of using such hacks inside test

Reply:
This is not a hack.
There are 2 enum clases: op::v5::NonMaxSuppression::BoxEncodingType and op::v9::NonMaxSuppression::BoxEncodingType. And we cannot assign directly one to another

Reply:
The problem of using `v5` here is because the test uses `v5` at test instantiation. So I sking you to fix it on instantiation instead of checking there

Reply:
Let's leave as is because fix will require more code with fixes in test classes inheritance